x/poolmanager/hooks.go

```diff
@@ -0,0 +1,42 @@
+package poolmanager
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/osmoutils"
+	epochstypes "github.com/osmosis-labs/osmosis/x/epochs/types"
+)
+
+func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	return nil
+}
+
+// At the end of each epoch, set the denom pair routes.
+func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	_ = osmoutils.ApplyFuncIfNoError(ctx, func(cacheCtx sdk.Context) error {
+		_, err := k.SetDenomPairRoutes(cacheCtx)
```

Comment:
Have we benchmarked this change? (Asking since we're adding additional computation on our epoch)

Reply:
The computation takes approx 1 second at maximum